src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -176,61 +207,214 @@ def __blob_is_not_complete(metadata):
+                "'--source-location' should be a local directory path or remote url.")
+    else:
+        source_location = _check_remote_source_code(source_location)
 
-    # [Server Bug] we should allow skiping image_name if users don't want to push
     is_push_enabled = True
     if image_name is None:
         is_push_enabled = False
+        print("'--tag' is not providied. Skip image push after build.")
+    else:
+        image_name = _check_image_name(image_name)
 
     # hard-code platform to linux and cpu to 1
     platform = PlatformProperties("Linux")
     platform.cpu = 1
 
-    # figure out how to pass build_arguments
     build_arguments = []
-    if not (arguments is None):
-        for name_value in arguments:
+    if not (build_args is None):
+        for name_value in build_args:
             name, value = name_value.split('=', 1)
             build_arguments.append(BuildArgument(name, value, False))
 
-    if not (secret_arguments is None):
-        for name_value in secret_arguments:
+    if not (secret_build_args is None):
+        for name_value in secret_build_args:
             name, value = name_value.split('=', 1)
             build_arguments.append(BuildArgument(name, value, True))
 
-    build_request = QueueBuildRequest(
-        image_name=image_name,
-        source_location=source_location,
-        build_parameters=build_parameters,
-        is_push_enabled=is_push_enabled,
-        timeout=timeout,
-        platform=platform,
-        build_arguments=build_arguments)
-
-    result = LongRunningOperation(cmd.cli_ctx)(client.queue(
-        build_request=build_request, resource_group_name=resource_group_name, registry_name=registry_name))
-
-    if isinstance(result, Build):
-        print("Successfully queued a build with build-id: {}. Starting to stream the logs...".format(result.build_id))
-        return acr_build_show_logs(cmd, client, registry_name, result.build_id, resource_group_name)
+    try:
+        build_request = QueueBuildRequest(
+            image_name=image_name,
+            source_location=source_location,
+            build_parameters=build_parameters,
+            is_push_enabled=is_push_enabled,
+            timeout=timeout,
+            platform=platform,
+            build_arguments=build_arguments)
+
+        result = LongRunningOperation(cmd.cli_ctx)(client.queue(
+            build_request=build_request, resource_group_name=resource_group_name, registry_name=registry_name))
+
+        print("Successfully queued a build with build-id: {}.".format(result.build_id))
+
+        if show_logs == 'true':
+            print("Starting to stream the logs...")
+            return acr_build_show_logs(cmd, client, registry_name, result.build_id, resource_group_name)
+    except Exception as err:
+        raise CLIError(err)
+
+
+def _check_remote_source_code(source_location):
+
+    # NOTE: acr_build doesn't support git repo url
+    if source_location.lower().startswith("git://"):
```

Comment:
I think we probably want to support the others too (`git@`, `git://`). For example you should be able to do `git@bitbucket.org:atlassianlabs/atlassian-docker.git` which would be different than github - same thing for lots of other providers

Reply:
At this point I think we can assume anything that isn't a sas url pattern is a git repo. 

---
In reply to: [168360092](https://github.com/AppPlatPerf/azure-cli/pull/60#discussion_r168360092) [](ancestors = 168360092)